aiidalab_widgets_base/computers.py

```diff
@@ -67,50 +65,53 @@ def __init__(self, **kwargs):
                                                disabled=False)
         self._verification_mode.observe(self.on_use_verification_mode_change, names='value')
 
-        # Proxy ssh settings
-        self._use_proxy = ipw.Checkbox(value=False, description='Use proxy')
-        self._use_proxy.observe(self.on_use_proxy_change, names='value')
-        self._inp_proxy_address = ipw.Text(description="Proxy server address:",
-                                           layout=ipw.Layout(width="350px"),
-                                           style=STYLE)
+        # Proxy ssh settings.
+        inp_use_proxy = ipw.Checkbox(value=False, description='Use proxy')
+        inp_use_proxy.observe(self.on_use_proxy_change, names='value')
+        link((inp_use_proxy, 'value'), (self, 'use_proxy'))
+
+        inp_proxy_hostname = ipw.Text(description="Proxy server address:",
+                                      layout=ipw.Layout(width="350px"),
+                                      style=STYLE)
+        link((inp_proxy_hostname, 'value'), (self, 'proxy_hostname'))
+
         self._use_diff_proxy_username = ipw.Checkbox(value=False,
                                                      description='Use different username and password',
                                                      layout={'width': 'initial'})
         self._use_diff_proxy_username.observe(self.on_use_diff_proxy_username_change, names='value')
-        self._inp_proxy_username = ipw.Text(value='',
-                                            description="Proxy server username:",
-                                            layout=ipw.Layout(width="350px"),
-                                            style=STYLE)
+        inp_proxy_username = ipw.Text(value='',
+                                      description="Proxy server username:",
+                                      layout=ipw.Layout(width="350px"),
+                                      style=STYLE)
+        link((inp_proxy_username, 'value'), (self, 'proxy_username'))
+
         self._inp_proxy_password = ipw.Password(value='',
```

Comment:
Any reason why this is a private member, but `inp_proxy_username` for example is not?

Reply:
I keep `_inp_proxy_password` as a class member because I access it later. It is private for the obvious reason - I want to prevent other classes to rely on it.

`inp_proxy_username` is not even a class member, as I don't want to expose the widget for the modifications.